### PR TITLE
Fix leftover compilation warnings

### PR DESCRIPTION
**Pull request by rerrahkr**
#218
Try to remove several compilation errors left from #221.

### CODE REVIEW

File: BambooTracker/midi/RtMidi/RtMidi.pri [hunk 1]
```diff
@@ -5,6 +5,10 @@ HEADERS += \
     $$PWD/RtMidi.hpp
 
 linux {
+    contains(DEFINES, __UNIX_JACK__) {
+        DEFINES += JACK_HAS_PORT_RENAME
```
Comment (OPNA2608) [inline BambooTracker/midi/RtMidi/RtMidi.pri:9]: I believe this only works if we limit what JACK version we support to JACK2. This should pose a problem on macOS where the commonly used JACK version from Homebrew is actually JACK1.

If we really can't detect the JACK version with qmake in the same way RtMidi's build systems do via header inspection or some other way, then maybe we should start thinking about making custom defines for convenience. `-D_USE_JACK1`, `-D_USE_JACK2`, `-D_USE_PULSE` etc.

Comment (rerrahkr) [reply BambooTracker/midi/RtMidi/RtMidi.pri:9]: I find that [`CONFIG` variable and qmake condition used it](https://doc.qt.io/qt-5/qmake-language.html#configuration-and-scopes).
How about making the `CONFIG` options `use_pulse`, `use_jack1` and `use_jack2` and add each `DEFINES` in these condition?

Comment (OPNA2608) [reply BambooTracker/midi/RtMidi/RtMidi.pri:9]: That looks like a better idea, yeah.

**Comment by OPNA2608** [comment]
About the build failure on Ubuntu, try apt package `libjack-jackd2-dev` instead of `libjack-dev`.

**Comment by OPNA2608** [comment]
So I actually read abit more into JACK now and it's abit more complicated than JACK1 vs JACK2.

Both JACK1 and JACK2 use the same header files for compilation, the daemons at runtime handle the version stuff on their own.

JACK1 0.125 and JACK2 both have these function declarations in `jack.h`:
https://github.com/jackaudio/headers/blob/eb67fa5ffc915203cdc8757572523c1b036840d9/jack.h#L759-L778
```cpp
int jack_port_set_name (jack_port_t *port, const char *port_name) JACK_OPTIONAL_WEAK_DEPRECATED_EXPORT;
int jack_port_rename (jack_client_t* client, jack_port_t *port, const char *port_name) JACK_OPTIONAL_WEAK_EXPORT;
```
Hence compilation on macOS with `-DJACK_HAS_PORT_RENAME` actually worked!

The reason it initially didn't work on Ubuntu is because Ubuntu 16.04 actually has JACK1 0.12**4**, which hadn't introduced the `jack_port_rename` replacement yet:
https://github.com/jackaudio/headers/blob/a14583ed99d346dacd6579d2d06e273a0c834b24/jack.h#L759-L766
```cpp
int jack_port_set_name (jack_port_t *port, const char *port_name) JACK_OPTIONAL_WEAK_EXPORT;
```
Thus it failed when trying to use the `rename` replacement.
```
midi/RtMidi/RtMidi.cpp: In member function ‘virtual void MidiInJack::setPortName(const string&)’:
midi/RtMidi/RtMidi.cpp:3115:64: error: ‘jack_port_rename’ was not declared in this scope
  jack_port_rename( data->client, data->port, portName.c_str() );
                                                               ^
```

---

So what we actually need is something like `use_jack` and `jack_has_rename`. If `use_jack` is set, setup all the JACK stuff. Additionally if `jack_has_rename` is set, add `DEFINES += JACK_HAS_PORT_RENAME`. Sorry for all the confusion.

**Comment by OPNA2608** [comment]
LGTM, just needs documentation updates. See Readme section `Building`, specifically

- `Dependencies -> Linux / BSD -> Debian / Ubuntu`,
the `libjack-dev` to `libjack-jackd2-dev # for JACK2, or libjack-dev for JACK1`

- `Compilation`,
all the `DEFINES+=` to the corresponding `CONFIG+=` equivalents

**Comment by OPNA2608** [comment]
One last nitpick, explain when dropping `CONFIG+=jack_has_rename` might be needed:

> If you know that building with an old version of JACK that lacks the `jack_port_rename` method or you encounter build problems like
> 
> > error: ‘jack_port_rename’ was not declared in this scope
> 
> , try dropping `CONFIG+=jack_has_rename`.

**Comment by rerrahkr** [comment]
All fixed. I will merge this later.

**Comment by rerrahkr** [comment]
Thanks for your help!

**Pull request merged**